src/actions/copy-environment/index.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import * as core from '@actions/core';
+import { DefaultRunnerFactory, RunnerFactory } from '../../lib';
+
+(async () => {
+    if (process.env.GITHUB_ACTIONS) {
+        await main(DefaultRunnerFactory);
+    }
+})();
+
+export async function main(factory: RunnerFactory): Promise<void> {
+    try {
+        core.startGroup('copy-environment:');
+        const username = core.getInput('user-name', { required: true });
+        const password = core.getInput('password-secret', { required: true });
+        if (!password || password.length === 0) {
+            return core.setFailed('Missing password! Specify one by setting input: \'password-secret\'');
+        }
+
+        const sourceUrl = core.getInput('source-url', { required: true});
+        const targetUrl = core.getInput('target-url', { required: true});
+
+        const pac = factory.getRunner('pac', process.cwd());
+        await pac.run(['auth', 'clear']);
+        await pac.run(['auth', 'create', '--kind', 'ADMIN', '--username', username, '--password', password]);
+
+        const copyEnvironmentArgs = ['admin', 'copy', '--source-url', sourceUrl, '--target-url', targetUrl];
+        const result = await pac.run(copyEnvironmentArgs);
```

Comment:
I really need to fix lint to be a build error :-) here's another one

Reply:
I am sorry, I didn't see this one on my vscode. I'll fix it.